main/main.cpp

```diff
@@ -4905,6 +4909,8 @@ void Main::cleanup(bool p_force) {
 	// Flush before uninitializing the scene, but delete the MessageQueue as late as possible.
 	message_queue->flush();
 
+	GDExtensionManager::get_singleton()->shutdown();
```

Comment:
And this one I would move a little earlier. Maybe before cleaning up the `TextServerManager`'s interfaces?